### DISCUSSION TITLE
Change costs / incompatibilities between vectors and span-views.

### DISCUSSION THREAD

**Discussion by phoeinx** [hyrise-mp-22-23/hyrise#114] [Ideas]
This thread is collecting known incompatibilities/limitations for our codebase when switching from vector-based to span-based access to data. 

Std::span offers a far more limited interface than std::vector (but a strict subset), so we will probably be unable to adapt everything identically. This thread documents incompatibilities and invites discussion of the relevance/impact of changes.

**Comment by phoeinx** [comment]
First incompatibility: No `cbegin`/`cend` iterators.

E.g. encountered in `dictionary_segment_iterable.hpp` making the following change necessary:

```cpp
auto begin = PointAccessIterator<Decompressor, DictionaryIteratorType, PosListIteratorType>{
           _dictionary->cbegin(), _segment.null_value_id(), vector.create_decompressor(), position_filter->cbegin(),
           position_filter->cbegin()};
```

```cpp
auto begin = PointAccessIterator<Decompressor, DictionaryIteratorType, PosListIteratorType>{
           _dictionary_span->begin(), _segment.null_value_id(), vector.create_decompressor(), position_filter->cbegin(),position_filter->cbegin()};
```


Do we know if using begin() instead of cbegin() / normal iterator instead of const iterator will have any impact on us? (I know that std::span doesn't offer cbegin, just asking if someone already thought about it - but would be fine with running with it for now. I think there was some discussion about offering a cbegin/cend interface for std::spans but would have to research what's the current state.)
@Bouncner perhaps?

**Reply by Bouncner** [reply]
Just remove the const iterators for now. Should be fine for now.

**Reply by Bouncner** [reply]
Are the segment spans all `const`?
